src/components/data/data-utils.js

```diff
@@ -340,8 +331,9 @@ export const EntityUploader = ({ onSuccess, onDismiss, namespace, name, entityTy
   }
 
   const match = /(?:membership|entity):([^\s]+)_id/.exec(fileContents)
-  const isInvalid = isFileImportCurrMode === isFileImportLastUsedMode && file && !match
+  const isInvalid = dataProvider.tsvFeatures.isInvalid(isFileImportCurrMode === isFileImportLastUsedMode, file, !match, fileContents.split('\n')[0].match(/sys_name/)) //only look at the first line
```

Comment:
Not sure how we might feel about this, but we have no validation that the file type is of `tsv` -- figure this is probably out-of-scope, but perhaps worth mentioning

Reply:
Hmm.  It'll fail to import and tell you `"We could not parse any data rows in your tsv file."` if you give it a non-tsv file (that otherwise meets criteria) - is that clear enough to the user or is it a better experience (and something we need to look out for) to have it prevent uploading the file in the first place?

Reply:
That's true -- probably enough to help the user correct their error/double-check their file type.